contracts/src/v0.8/automation/dev/v2_3/AutomationRegistryBase2_3.sol

```diff
@@ -953,4 +969,36 @@ abstract contract AutomationRegistryBase2_3 is ConfirmedOwner {
       revert OnlySimulatedBackend();
     }
   }
+
+  /**
+   * @notice sets billing configuration for a token
+   * @param billingTokens the addresses of tokens
+   * @param billingConfigs the configs for tokens
+   */
+  function _setBillingConfig(address[] memory billingTokens, BillingConfig[] memory billingConfigs) internal {
+    // Clear existing data
+    for (uint256 i = 0; i < s_billingTokens.length; i++) {
+      delete s_billingConfigs[s_billingTokens[i]];
+    }
+    delete s_billingTokens;
+
+    for (uint256 idx = 0; idx < billingTokens.length; idx++) {
+      address token = billingTokens[idx];
+      BillingConfig memory config = billingConfigs[idx];
+
+      if (token == address(0) || config.priceFeed == address(0)) {
+        revert ZeroAddressNotAllowed();
+      }
+
+      // if this is a new token, add it to tokens list
+      if (s_billingConfigs[token].priceFeed == address(0)) {
+        s_billingTokens.push(token);
+      }
```

Comment:
can we have an `else { revert DuplicateEntry() }` here?

Reply:
or maybe simpler...
```
if (s_billingConfigs[token].priceFeed != address(0)) {
  revert DuplicateEntry();
}
s_billingTokens.push(token);
```

Reply:
Got it, we will not swallow the duplicate entries, we will revert!